### ISSUE TITLE
slatedocs/slate not found

### ISSUE THREAD

**Issue by yangnom** [bep/docuapi#51]
I do the command

`hugo server -t docuapi`

and it installs some modules and then apparently tries to install the modeul for 'slatedocs/slate' but then fails with this erro

`Error: module "github.com/slatedocs/slate" not found; either add it as a Hugo Module or store it in "/Users/yangnom/www/quickstart/themes".: module does not exist'

I'm guessing because that reference doesn't exist anymore, but maybe I'm just dumb and doing something wrong...

**Comment by eine** [comment]
Do you have a `go.mod` file as in the example? https://github.com/bep/docuapi/blob/master/exampleSite/go.mod

**Comment by yangnom** [comment]
This would be my `go.mod` file

```
module github.com/bep/docuapi

go 1.13

require (
	github.com/bep/empty-hugo-module v1.0.0 // indirect
	github.com/jquery/jquery-dist v0.0.0-20191202185020-55b7d19f7bf1 // indirect
	github.com/olivernn/lunr.js v2.3.8+incompatible // indirect
	github.com/slatedocs/slate v2.4.0+incompatible // indirect
)

```

**Comment by eine** [comment]
It seems good. This is the one I'm using: https://github.com/eine/issue-runner/blob/master/doc/go.mod. And this is a GitHub Actions workflow that may be useful for you to debug your issue in a "clean" environment: https://github.com/eine/issue-runner/blob/master/.github/workflows/doc.yml.

**Comment by yangnom** [comment]
@eine thanks very much for your help!

I'm just starting with hugo and messing around a bit and docuapi looks /hot/ so I just wanted to try it real quickly, so I'm probably not going to spend a lot of time trying to figure it out yet (I can barely use hugo as it is), but I really appreciate your help and will probably come back to this in the future!

**Comment by davidmarcielp** [comment]
I am having the same issue, 
Error: module "github.com/slatedocs/slate" not found; either add it as a Hugo Module or store it in "/home/david/Documents/quickstart/themes".: module does not exist
I dont find the way to make it work

**Comment by bep** [comment]
@davidmarcielp I could probably vendor these dependencies, but currently you need Go + Hugo installed for this to work. I admit that the error message could be better.

**Comment by davidmarcielp** [comment]
I have updated to the latest versions

> Hugo Static Site Generator v0.64.1-C327E75D linux/amd64 BuildDate: 2020-02-09T20:47:32Z

> go version go1.10.4 linux/amd64

I have also launched npm install inside themes/slate
but still the same issue:

> Error: module "github.com/slatedocs/slate" not found; either add it as a Hugo Module or store it in "/home/david/Documents/Hugo/quickstart slate/themes".: module does not exist

**Comment by eine** [comment]
@davidmarcielp, go 1.10 is EOL. Use 1.13 instead.

**Comment by nextdude-mdsol** [comment]
I'm still having this problem. I've got `go 1.14` and `hugo 0.70` and `go.mod` in place as per [eine](https://github.com/bep/docuapi/issues/51#issuecomment-572428507).

```zsh
➜  docs git:(data-gen) ✗ hugo
hugo: collected modules in 797 ms
Total in 802 ms
Error: module "github.com/slatedocs/slate" not found; either add it as a Hugo Module or store it in "/Users/nextdude/Google_Drive/epiphanous/medidata/git/meds-connected-devices/graphql/docs/themes".: module does not exist
➜  docs git:(data-gen) ✗ go version
go version go1.14.2 darwin/amd64
➜  docs git:(data-gen) ✗ hugo version
Hugo Static Site Generator v0.70.0/extended darwin/amd64 BuildDate: unknown
➜  docs git:(data-gen) ✗
```

**Comment by eine** [comment]
@nextdude-mdsol, the workflow I referenced above is still working fine (it run some minutes ago). Hence, I would suggest to copy it and make the changes you need to adapt it to your use case:

- Update hugo to v0.70.0.
- Use `macos-latest` instead of `ubuntu-latest`.

**Comment by wsamoht** [comment]
For anyone else having an issue this is what you need to do.

Created `go.mod` with the following:
```go
module github.com/bep/docuapi

go 1.12

require (
	github.com/bep/empty-hugo-module v1.0.0 // indirect
	github.com/jquery/jquery-dist v0.0.0-20190501211928-15bc73803f76 // indirect
	github.com/olivernn/lunr.js v2.3.8+incompatible // indirect
	github.com/slatedocs/slate v2.3.1+incompatible // indirect
)
```

Then in the config.toml make sure you have the following:
```
[module]
[module.hugoVersion]
[[module.imports]]
# We need one module import for Hugo to detect us as a Hugo Module.
path="github.com/bep/empty-hugo-module"
```

**Comment by theonlijuan** [comment]
> For anyone else having an issue this is what you need to do.
> 
> Created `go.mod` with the following:

Hi, newbie here. Where do I create the `go.mod`?

**Comment by wsamoht** [comment]
`go.mod` goes in the root directory alongside `config.toml`.

**Comment by ctfrancia** [comment]
update for 2025
in your root path of Hugo project your **hugo**.toml file should look like this, if you are starting a brand new project
```toml
baseURL = 'https://example.org/'
languageCode = 'en-us'
title = 'My New Hugo Site'
theme = 'docuapi'
[[module.imports]]
path = "github.com/bep/docuapi/v2"
``` 
on the same level as that `hugo.toml` file you should have your `go.mod` path, if you do *not* have that then  I suggest entering this command in your terminal `go mod init github.com/<YOUR_USERNAME>/<YOURPROJECT>` mine in this example is `go mod init github.com/ctfrancia/buho-api-docs`.

tldr:
in root path:
1. create a go module with `go mod init github.com/<YOUR_USERNAME>/<YOURPROJECT>`
2. inside your `hugo.toml` file at the bottom add the [module.imports](https://github.com/bep/docuapi?tab=readme-ov-file#use) like stated in their docs
3. in terminal type `go mod tidy` 
4. prosper

that should be enough to run `hugo server` now
